### PR TITLE
Feature parity for Hub role creation, edit and details

### PR DESCRIPTION
**Pull request by vidyanambiar**
## Existing UI

image

## New UI

image

### Role details

The existing UI does not have a role details page but this was added to keep the roles UI consistent across components.

image

### CODE REVIEW

File: frontend/hub/access/roles/Roles.tsx [hunk 1]
```diff
@@ -29,7 +28,7 @@ export function HubRolesTable() {
 
   const view = usePulpView<Role>({
     url: pulpAPI`/roles/`,
-    keyFn: nameKeyFn,
+    keyFn: (role: { pulp_href: string }) => role.pulp_href,
```
Comment (jamestalton) [inline frontend/hub/access/roles/Roles.tsx:31]: This line might cause issues as the keyFn needs to not change between rerenders.

**Review by nixocio — COMMENTED** [review]
@vidyanambiar I got this code locally, and it works great. 💯 

I left a couple of minor suggestions, and questions.

File: frontend/hub/access/roles/hooks/useRoleActions.tsx [hunk 2]
```diff
@@ -68,8 +70,11 @@ export function useRoleRowActions(onComplete: (roles: Role[]) => void) {
             : t(
                 'You do not have permission to edit this role. Please contact your organization administrator if there is an issue with your access.'
               ),
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        onClick: (role) => alert('TODO'),
+        href: (role) => {
+          return getPageUrl(HubRoute.EditRole, {
+            params: { id: parsePulpIDFromURL(role.pulp_href) ?? '' },
```
[the remark this answers is not in the document]
Comment (vidyanambiar) [reply frontend/hub/access/roles/hooks/useRoleActions.tsx]: Thanks for catching that. Looks like role names are unique so I updated the code to use name instead.

File: frontend/hub/access/roles/RolePage/RoleForm.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  const postRequest = usePostRequest<RoleRequestBody>();
+  const onSubmit: PageFormSubmitHandler<RoleInput> = async (values) => {
+    const { name, description, permissionCategories } = values;
+    const permissions = getSelectedPermissions(permissionCategories);
+    await postRequest(pulpAPI`/roles/`, {
+      name,
+      description,
+      permissions,
+    });
+    navigate(-1);
```
Comment (nixocio) [inline frontend/hub/access/roles/RolePage/RoleForm.tsx:46]: I just noticed that one now. I believe we are redirecting to the details page after creating resources in general.

**Review by nixocio — APPROVED** [review]
LGTM

File: frontend/hub/access/roles/RolePage/RoleForm.tsx (hunk 3 above)
Comment (vidyanambiar) [reply frontend/hub/access/roles/RolePage/RoleForm.tsx]: updated 👍

**Pull request merged**